src/level/level.cpp

```diff
@@ -225,18 +225,15 @@ namespace level {
     void Level::enemiesAttack(WINDOW *win, manager::Level *levelManager) {
         Position attackPos;
         for (unsigned int i = 0; i < enemies_.size(); i++) {
-            for (int j = (enemies_[i]->getPosition().riga) - 1; j <= (enemies_[i]->getPosition().riga) + 1; j++) {
-                for (int k = (enemies_[i]->getPosition().colonna) - 1; k <= (enemies_[i]->getPosition().colonna) + 1; k++) {
-                    attackPos = Position(j, k);
-                    if ((levelManager->getPlayer()->getPosition()) == attackPos) {
-                        enemies_[i]->attack(levelManager->getPlayer());
-                        if (levelManager->getPlayer()->isDead()) {
-                            // TODO (gio?): implementare una finestra di gameover
-                        }
-                        enemies_[i]->clear(win);
-                        delete enemies_[i];
-                        enemies_.remove(i);
+            if((levelManager->getPlayer()->getPosition().riga - enemies_[i]->getPosition().riga) <= 1 && (enemies_[i]->getPosition().riga- levelManager->getPlayer()->getPosition().riga) >= -1){
+                if((levelManager->getPlayer()->getPosition().colonna - enemies_[i]->getPosition().colonna) <= 1 && (enemies_[i]->getPosition().colonna- levelManager->getPlayer()->getPosition().colonna) >= -1){
+                    enemies_[i]->attack(levelManager->getPlayer());
+                    if (levelManager->getPlayer()->isDead()) {
+                        // TODO (gio?): implementare una finestra di gameover
                     }
+                    enemies_[i]->clear(win);
+                    delete enemies_[i];
+                    enemies_.remove(i);
```

Comment:
Ora, non ne sono sicuro, ma secondo me è buggata questa implementazione. 

Il remove muove l'ultimo enemy del vector in posizione i, e riduce le dimensioni, e poi col for vai avanti con la i, andando subito in posizione `i + 1`, quindi il nuovo enemy in posizione i non la vai mai a calcolare.

Oltre a ciò i due if che vanno a calcolare la distanza potrebbero essere uno solo.